tests/ui_tools/test_messages.py

```diff
@@ -1585,14 +1587,120 @@ def test_keypress_EDIT_MESSAGE(
             # fmt: on
         ],
     )
-    def test_transform_content(self, mocker, raw_html, expected_content):
+    def test_transform_content(self, raw_html: str, expected_content: Any) -> None:
         expected_content = expected_content.replace("{}", QUOTED_TEXT_MARKER)
 
         content, *_ = MessageBox.transform_content(raw_html, SERVER_URL)
 
         rendered_text = Text(content)
         assert rendered_text.text == expected_content
 
+    @pytest.mark.parametrize(
+        "raw_html, expected_message_links",
+        [
+            (
+                """
+                <p><a href="https://github.com/zulip/zulip-terminal/pull/1">https://github.com/zulip/zulip-terminal/pull/1</a></p>
+                """,
+                {
+                    "https://github.com/zulip/zulip-terminal/pull/1": (
+                        "github.com",
+                        1,
+                        True,
+                    )
+                },
+            ),
+            (
+                """
+                <p><a href="https://foo.com">https://foo.com</a></p>
+                """,
+                {"https://foo.com": ("https://foo.com", 1, False)},
+            ),
+            (
+                """
+                <p><a href="#narrow/stream/206-zulip-terminal/topic/announce">https://chat.zulip.zulip/#narrow/stream/206-zulip-terminal/topic/announce</a></p>
```

Comment:
Are these test cases all from real example messages, to indicate this is what we actually receive?